include/node.tcc

```diff
@@ -537,6 +583,23 @@ bool mpm::Node<Tdim, Tdof, Tnphases>::mpi_rank(unsigned rank) {
   return status.second;
 }
 
+//! Compute mass density (Z. Wiezckowski, 2004)
+//! density = mass / lumped volume
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+void mpm::Node<Tdim, Tdof, Tnphases>::compute_density() {
+  const double tolerance = 1.E-16;  // std::numeric_limits<double>::lowest();
+
+  for (unsigned phase = 0; phase < Tnphases; ++phase) {
+    if (mass_(phase) > tolerance) {
+      if (volume_(phase) > tolerance)
+        density_(phase) = mass_(phase) / volume_(phase);
+
+      // Check to see if value is below threshold
+      if (std::abs(density_(phase)) < 1.E-15) density_(phase) = 0.;
```

Comment:
Why density is checked against 1e-15 while the rest is 1e-16?

Reply:
I think there is nothing different, I have change it to the same tolerance.